### PR TITLE
Vagrant: Configurable # of VMs, ansible_local provisioner

### LINKED ISSUE #256: more simulated vagrant up
It would be great to have vagrant up more simulate networked data centers. Should have configurable amount of data centers, within each data center (virtual ip e.g. https://github.com/mesos/kafka/blob/master/src/vagrant/Vagrantfile#L18 ) configurable amount of masters and slaves. The defaults for each would be == 1. This way we can test and launch on the two different services the clusters and operations between them more succinctly from our host machine or just role a single instance.

### PR DESCRIPTION
**Pull request by siddharthist**
This PR:
 * Enables a configurable number of VMs
 * Removes load_security from Vagrantfile
 * Uses the new [Vagrant 1.8](https://docs.vagrantup.com/v2/provisioning/ansible_local.html) [ansible_local provisioner](https://docs.vagrantup.com/v2/provisioning/ansible_local.html)
 * Creates an inventory for Ansible entirely within Vagrant, without external tools
 * Thoroughly documents our Vagrant setup

It fixes #256 (more completely than #929), #499, #674, #888, #930, #941 and makes progress on #877 (with work from #929).

I suggest we either merge #929 now and then this PR, or just close #929 and focus on this one. Its work is entirely reproduced or enhanced here. 

I have tested it with varying numbers of workers, but don't have the RAM to test 3 control nodes. It'd be great if anybody could test that.

Next steps:
 1. GlusterFS support
 2. LVM support
 3. Add edge nodes
 4. Update packer scripts/mi box (@ChrisAubuchon)
 5. Fix this very rare error if it still happens (unclear): http://pastebin.com/csrNsK07
 6. Multi-datacenter simulation
 7. Instead of `addons`, just have a list of playbooks to run with the default being `[terraform.sample.yml]`, so you can use `vagrant provision` to run other playbooks.

### CODE REVIEW

**Comment by siddharthist** [comment]
The examples documentation provided by #906 will need to be updated after this is merged.

**Comment by siddharthist** [comment]
Please note that this branch is currently affected by https://github.com/mitchellh/vagrant/issues/6763

File: Vagrantfile [hunk 1]
```diff
@@ -1,37 +1,122 @@
...
+require "yaml"
 
-def load_security
-  fname = File.join(File.dirname(__FILE__), "security.yml")
-  if !File.exist? fname
-    $stderr.puts "security.yml not found - please run `./security-setup` and try again."
-    exit 1
-  end
+# Number of VMs to create
+WORKERS=1
```
Comment (ryane) [inline Vagrantfile]: could we add support for optionally reading these variables (and other variables like cpu/memory) from an external config file that can be gitignored? that way you can maintain a custom vagrant setup without a dirty working tree

Comment (siddharthist) [inline Vagrantfile]: It would have to be an optional config file with sensible details, so users can still just `vagrant up`, but yeah, absolutely!

Comment (ryane) [inline Vagrantfile]: yep, definitely should be optional

File: Vagrantfile [hunk 2]
```diff
@@ -1,37 +1,133 @@
...
+ num_controls = config_hash.fetch("num_controls", num_controls)
+ worker_ip_start = config_hash.fetch("worker_ip_start", worker_ip_start)
+ control_ip_start = config_hash.fetch("control_ip_start", control_ip_start)
 end
 
-VAGRANT_PRIVATE_IP = "192.168.242.55"
 
-Vagrant.configure(2) do |config|
+# Check Vagrant version before continuing. TODO: delete when 2.0 is out.
+if not `vagrant --version` =~ /(1\.([89]|1[0-9])|2\.[0-9])/
```
Comment (rygwdn) [inline Vagrantfile]: This could be replaced with `Vagrant.require_version ">= 1.8"`

Comment (siddharthist) [inline Vagrantfile]: Oh nice! That's really convenient! Thanks @rygwdn

**Comment by ryane** [comment]
@siddharthist good call. readme updated

**Comment by ohal** [comment]
@ohal to jenkins
test this please
ok to test

**Comment by siddharthist** [comment]
All of @ryane's changes look good to me! Can you test this @sehqlr?

**Comment by sehqlr** [comment]
I can test this, and I'll post an update here

**Comment by siddharthist** [comment]
@sehqlr Has tested this and confirmed that it works.

**Pull request merged**